sql/parser/normalize.go

```diff
@@ -99,6 +99,15 @@ func (v *normalizeVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
 
 	// Evaluate constant expressions.
 	if isConst(expr) {
+		// Normalize constant tuples present in In and NotIn comparison expressions
+		if cmp, ok := expr.(*ComparisonExpr); ok {
```

Comment:
This isn't only normalizing constant tuples present in `{IN,NOT IN}` expressions which are themselves constant. I think you also need to add something to `normalizeComparisonExpr` for when `isVar(ComparisonExpr.Left) && isConst(ComparisonExpr.Right)`.